xml_converter/generators/cpp_templates/class_template.cpp

```diff
@@ -154,11 +165,23 @@ waypoint::{{cpp_class}} {{cpp_class}}::as_protobuf() const {
     if (this->{{attribute_variable.attribute_name}}_is_set) {
         proto_{{cpp_class_header}}.set_{{attribute_variable.protobuf_field}}(to_proto_{{attribute_variable.class_name}}(this->{{attribute_variable.attribute_name}}));
     }
-    {% elif (attribute_variable.attribute_type in ["MultiflagValue", "CompoundValue", "Custom"]) and attribute_variable.is_child == false%}
+    {% elif (attribute_variable.attribute_type in ["MultiflagValue", "CompoundValue", "Custom"]) and attribute_variable.compound_name == None%}
+    {% if (attribute_variable.class_name == "color")%}
+    if (this->{{attribute_variable.attribute_name}}_is_set) {
+        if (this->alpha_is_set) {
+            proto_{{cpp_class_header}}.set_allocated_rgba(to_proto_color(this->color, this->color.alpha));
+        }
+        else {
+            proto_{{cpp_class_header}}.set_allocated_rgba(to_proto_color(this->color, 1.0));
+        }
+    }
+    {% else: %}
     if (this->{{attribute_variable.attribute_name}}_is_set) {
         proto_{{cpp_class_header}}.set_allocated_{{attribute_variable.protobuf_field}}(to_proto_{{attribute_variable.class_name}}(this->{{attribute_variable.attribute_name}}));
     }
-    {% elif attribute_variable.is_child == true%}
+    {% endif %}
+    {% elif (attribute_variable.compound_name != None)%}
+    {% elif (attribute_variable.attribute_name == "alpha")%}
```

Comment:
Don't branch on arbitrary documentation strings. This would mean that if someone wanted to change the documentation they might also need to change this template file too.

Reply:
Understood. Reverting this.